proto/ignition/msgs/particle_emiter.proto

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package ignition.msgs;
+option java_package = "com.ignition.msgs";
+option java_outer_classname = "ParticleEmitterProtos";
+
+/// \ingroup ignition.msgs
+/// \interface ParticleEmitter
+/// \brief Message for a particle emitter.
+
+import "ignition/msgs/color.proto";
+import "ignition/msgs/header.proto";
+import "ignition/msgs/material.proto";
+import "ignition/msgs/pose.proto";
+import "ignition/msgs/vector3d.proto";
+
+message ParticleEmitter
+{
+  /// \brief Optional header data.
+  Header header                 = 1;
+
+  /// \brief The emitter name.
+  string name                   = 2;
+
+  /// \brief All possible emitter types.
+  enum EmitterType
+  {
+    /// \brief Point emitter.
+    POINT     = 0;
+    /// \brief Box emitter.
+    BOX       = 1;
+    /// \brief Cylinder emitter.
+    CYLINDER  = 2;
+    /// \brief Ellipsoid emitter.
+    ELLIPSOID = 3;
+  }
+  /// \brief The emitter type.
+  EmitterType type              = 3;
+
+  /// \brief The position of the emitter.
+  Pose pose                     = 4;
+
+  /// The size of the emitter where the particles are sampled.
+  Vector3d size                 = 5;
+
+  /// \brief How many particles per second should be emitted.
+  float rate                    = 6;
+
+  /// \brief The number of seconds the emitter is active.
+  float duration                = 7;
+
+  /// \brief Whether particle emitter is enabled or not.
+  bool emitting                 = 8;
+
+  /// \brief The particle dimensions (width, height, depth).
+  Vector3d particle_size        = 9;
+
+  /// \brief The number of seconds each particle will ’live’ for before
+  /// being destroyed.
+  float lifetime                = 10;
+
+  /// \brief The material which all particles in the emitter will use.
+  Material material             = 11;
+
+  /// \brief The minimum velocity each particle is emitted (m/s).
+  float min_velocity            = 12;
+
+  /// \brief The maximum velocity each particle is emitted (m/s).
+  float max_velocity            = 13;
+
+  /// \brief The starting color of the particles.
+  Color color_start             = 14;
+
+  /// \brief The end color of the particles.
+  Color color_end               = 15;
+
+  /// \brief The amount by which to scale the particles in both x and y
+  /// direction per second.
```

Comment:
The size is in 3D, but the scale only mentions X and Y. Is this in screen coordinates?

Reply:
I agree, it's weird. This comes from Ogre, see [here](https://ogrecave.github.io/ogre-next/api/2.1/class_ogre_1_1_scale_affector.html#ad31d03a0135d19e30c0ef60bbdd3a90a). I suspect that the scaling happens in screen coordinates but it's not explained.

Reply:
Got it. Since we're supposed to support arbitrary rendering engines, I think it would be interesting to make it very clear what each parameter means so that they're used correctly outside Ogre.

Reply:
Added the note that the scaling works in screen coordinates. 2494142